types/dialogify-browser/Dialogify.d.ts

```diff
@@ -0,0 +1,238 @@
+         * @param this Dialogify instance.
+         */
+        cancel?: (this: Dialogify) => void;
+    }
+
+    interface PromptDialogOptions extends SimpleDialogOptions {
+        /** Placeholder in text input. */
+        placeholder?: string;
+        /**
+         * Callback function when dialog confirmed.
+         * @param this Dialogify instance.
+         * @param value User inputed value.
+         */
+        ok?: (this: Dialogify, value: string) => void;
+        /**
+         * Callback function when dialog dismissed
+         * @param this Dialogify instance.
+         */
+        cancel?: (this: Dialogify) => void;
+    }
+
+    interface DialogStyle {
+        /** CSS styles in key-value pair object to be used by dialog. */
+        style?: { [key: string]: string };
+        /** CSS class name to be used by dialog. */
+        className?: string;
+        /** CSS styles in key-value pair object to be used by dialog content. */
+        contentStyle?: { [key: string]: string };
+        /** CSS class name to be used by dialog content. */
+        contentClassName?: string;
+    }
+
+    interface ButtonStyle {
+        /** Image src to display as an icon. */
+        image?: string;
+        /** CSS styles in key-value pair object to be used by button. */
+        style?: { [key: string]: string };
+        /** CSS class name to be used by button. */
+        className?: string;
+    }
+
+    interface Button {
+        /**
+         * Button style, will auto close dialog if set to `BUTTON_PRIMARY`.
+         * Can be `Dialogify.BUTTON_PRIMARY`, `Dialogify.BUTTON_DANGER` or a class name.
+         */
+        type?: BUTTON_PRIMARY | BUTTON_DANGER | string;
+        /** Button text content, "關閉" by default. */
+        text?: string;
+        /**
+         * Callback function when button clicked
+         * @param this Dialogify instance.
+         * @param event Mouse click event
+         */
+        click?: (this: Dialogify, event: MouseEvent) => void;
+        /** Set `autofocus` property or not, `false` by default. */
+        focused?: boolean;
+        /** Set button as disabled or not, `false` by default. */
+        disabled?: boolean;
+        /** Set button id, if set, will also be key in `$buttonList`. */
+        id?: string;
+    }
+
+    interface ButtonImpl extends Button {
+        /** Enable the button. */
+        enable(): void;
+
+        /** Disable the button. */
+        disable(): void;
+    }
+
+    interface ButtonOption {
+        /**
+         * Button position style, right aligned by default.
+         * Can be `Dialogify.BUTTON_CENTER`, `Dialogify.BUTTON_LEFT` or a class name.
+         */
+        position: BUTTON_CENTER | BUTTON_LEFT | string;
+    }
+
+    type SIZE_LARGE = 'dialogify__autowidth';
+
+    type BUTTON_CENTER = 'text-center';
+    type BUTTON_LEFT = 'text-left';
+
+    type BUTTON_PRIMARY = 'btn-primary';
+    type BUTTON_DANGER = 'btn-danger';
```

Comment:
I would say those are const:
```suggestion
    const SIZE_LARGE: 'dialogify__autowidth';
    const BUTTON_CENTER: 'text-center';
    const BUTTON_LEFT: 'text-left';
    const BUTTON_PRIMARY: 'btn-primary';
    const BUTTON_DANGER: 'btn-danger';
```
https://github.com/OneupNetwork/dialogify/blob/master/src/js/dialogify.js#L259-L265

Reply:
and there should be `LOCALE` symbol keyed with `locale` keys

Reply:
Oh, disregard my comments about 'type' > 'const'. You've have this as static fields, which is perfectly OK. Locale symbol comment matters, if it can be used by users in some way.

Reply:
Got it, fixed!